bluetooth: CS: add bt_le_cs_get_antenna_path()

ivaniushkov opened the pull request:
added function to calculate antenna path
based on the antenna permutation index and
total number of antenna paths

jhedberg commented on subsys/bluetooth/host/cs.c:
```diff
@@ -1344,4 +1349,70 @@ void bt_le_cs_step_data_parse(struct net_buf_simple *step_data_buf,
+				  uint8_t antenna_path_permutation_index,
+				  uint8_t tone_index)
+{
+	if (n_ap == 2) {
+		return antenna_path_lut_n_ap_2[antenna_path_permutation_index][tone_index];
+	}
+	if (n_ap == 3) {
+		return antenna_path_lut_n_ap_3[antenna_path_permutation_index][tone_index];
+	}
+	if (n_ap == 4) {
+		return antenna_path_lut_n_ap_4[antenna_path_permutation_index][tone_index];
+	}
+	return 0;
```
I think this might look a bit better as a switch statement (something to consider whenever you have multiple if-statements testing the same variable):
```c
	switch (n_ap) {
	case 2:
		return antenna_path_lut_n_ap_2[antenna_path_permutation_index][tone_index];
	case 3:
		return ...
	...
	default:
		return 0;
	}
```

Btw, are you sure it's a good idea to return `A1` in the case of invalid inputs? Maybe a return value of `int` might be better, and then you return `-EINVAL` in case of invalid inputs?

ivaniushkov replied:
thanks, it makes sense. I added validation of arguments and started using switch-case instead of multiple if statements

olivier-le-sage commented on subsys/bluetooth/host/cs.c:
```diff
@@ -1344,4 +1349,108 @@ void bt_le_cs_step_data_parse(struct net_buf_simple *step_data_buf,
+/* Bluetooth Core Specification 6.0, Table 4.13, Antenna Path Permutation for N_AP=2.
+ * The last element corresponds to extension slot
+ */
+static uint8_t antenna_path_lut_n_ap_2[2][3] = {
```
should this and other LUTs be `static const` so the linker puts them in NVM ?

olivier-le-sage commented on subsys/bluetooth/host/cs.c:
```diff
@@ -1344,4 +1349,108 @@ void bt_le_cs_step_data_parse(struct net_buf_simple *step_data_buf,
+	{A2, A3, A4, A1, A1},
+};
+
+int bt_le_cs_get_antenna_path(uint8_t n_ap,
```
not sure if this makes this that more readable, but you could define:

```
#define FACTORIAL(n) ((n) <= 1 ? 1 : \
                 (n) == 2 ? 2 : \
                 (n) == 3 ? 6 : \
                 (n) == 4 ? 24 : INT_MAX)
```

then use this to simplify the code below by checking for invalid values before the switch-case.
Then you don't need comments to explain it.

can also use it to size the LUTs:

```
static uint8_t antenna_path_lut_n_ap_4[FACTORIAL(4)][5] = {
```

ivaniushkov committed:
bluetooth: CS: add bt_le_cs_get_antenna_path()

added function to calculate antenna path
based on the antenna permutation index and
total number of antenna paths

Signed-off-by: Ivan Iushkov <ivan.iushkov@nordicsemi.no>
